crates/primitives/src/transaction/mod.rs

```diff
@@ -345,9 +390,47 @@ impl Transaction {
             Transaction::Eip2930(TxEip2930 { input, .. }) |
             Transaction::Eip1559(TxEip1559 { input, .. }) |
             Transaction::Eip4844(TxEip4844 { input, .. }) => input,
+            #[cfg(feature = "optimism")]
+            Transaction::Deposit(TxDeposit { input, .. }) => input,
+        }
+    }
+
+    /// Returns the source hash of the transaction, which uniquely identifies its source.
+    /// If the transaction is not a deposit transaction, this will always return `B256::zero()`.
+    #[cfg(feature = "optimism")]
+    pub fn source_hash(&self) -> B256 {
```

Comment:
Make this an `Option<_>`, using default magic flag values is rough -- also keeps consistent with `mint()` below

Reply:
Addressed in https://github.com/anton-rs/op-reth/pull/161